deployer/utils.py

```diff
@@ -95,23 +99,23 @@ def verify_and_decrypt_file(encrypted_path):
         # We must then determine if the file is using sops
         # sops files are JSON/YAML with a `sops` key. So we first check
         # if the file is valid JSON/YAML, and then if it has a `sops` key
-        with open(encrypted_path) as f:
+        with open(original_filepath) as f:
 
             # Support the (clearly wrong) people who use .yml instead of .yaml
             if ext == ".yaml" or ext == ".yml":
                 try:
-                    encrypted_data = yaml.load(f)
+                    content = yaml.load(f)
                 except ScannerError:
-                    yield encrypted_path
+                    yield original_filepath
                     return
             elif ext == ".json":
                 try:
-                    encrypted_data = json.load(f)
+                    content = json.load(f)
                 except json.JSONDecodeError:
-                    yield encrypted_path
+                    yield original_filepath
                     return
 
-        if "sops" not in encrypted_data:
+        if "sops" not in content:
```

Comment:
I think right now, this function assumes our encrypted files are .json or .yaml - which is what we so far need. It doesn't indicate anything about this assumption in the docstring though, and it could be something of relevance in the future.

I think adding a hard failure with a FIXME note on being passed seemingly encrypted files that aren't .yml/.yaml/.json could make sense at this point.

Reply:
I feel like we are approaching feature creep on this PR now. This was opened as a bug fix so we can reenable deployments to three clusters that were broken after merging #1027. General improvements like this can be addressed in a new PR.

Reply:
Yepp =/ I felt that too, sorry - let's go for a merge!